Confusing error message when deposit is not made

### Description

When deploying a validator with teku and mevboost and the deposit is not made, it shows a very confusing error message on validator side. Consensus side and mevboost are both OK and show correct error messages related to a deposit that isn't yet made. 

### Steps to Reproduce (Bug)

Deploy a validator with mevboost and teku. (for EL I use Nethermind, but I assume this doesn't matter)

**Expected behavior:** Show an error message telling that the deposit is not yet made. Something like: 
`Failed to send validator registrations to Beacon Node | Deposit is not yet made`

**Actual behavior:** 
```
validator-client  | 2022-09-08 11:32:48.076 INFO  - 0 out of 1 validator(s) registrations were successfully sent to the Beacon Node.
validator-client  | 2022-09-08 11:32:48.077 ERROR - Validator   *** Failed to send validator registrations to Beacon Node
validator-client  | java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: Invalid params response from Beacon Node API (url = http://consensus:4000/eth/v1/validator/register_validator, status = 400, message = Status code 502 not supported)
validator-client  | at java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:145) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.whenComplete(Unknown Source) ~[?:?]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:580) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:32) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:142) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$exceptionallyCompose$30(SafeFuture.java:399) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:145) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.CompletableFuture.whenComplete(Unknown Source) ~[?:?]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:580) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:32) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:142) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateTo(SafeFuture.java:293) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:119) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
validator-client  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
validator-client  | at java.lang.Thread.run(Unknown Source) ~[?:?]
validator-client  | Caused by: java.lang.IllegalArgumentException: Invalid params response from Beacon Node API (url = http://consensus:4000/eth/v1/validator/register_validator, status = 400, message = Status code 502 not supported)
validator-client  | at tech.pegasys.teku.validator.remote.typedef.ResponseHandler.defaultBadRequestHandler(ResponseHandler.java:129) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.typedef.ResponseHandler.handleResponse(ResponseHandler.java:93) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.typedef.handlers.AbstractTypeDefRequest.executeCall(AbstractTypeDefRequest.java:149) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.typedef.handlers.AbstractTypeDefRequest.postJson(AbstractTypeDefRequest.java:127) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.typedef.handlers.RegisterValidatorsRequest.sendValidatorRegistrationsAsJson(RegisterValidatorsRequest.java:77) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.typedef.handlers.RegisterValidatorsRequest.registerValidators(RegisterValidatorsRequest.java:52) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.typedef.OkHttpValidatorTypeDefClient.registerValidators(OkHttpValidatorTypeDefClient.java:101) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.lambda$registerValidators$30(RemoteValidatorApiHandler.java:443) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.lambda$sendRequest$31(RemoteValidatorApiHandler.java:449) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:81) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.sendRequest(RemoteValidatorApiHandler.java:460) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.validator.remote.RemoteValidatorApiHandler.lambda$sendRequest$32(RemoteValidatorApiHandler.java:455) ~[teku-validator-remote-22.9.0.jar:22.9.0]
validator-client  | at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:73) ~[teku-infrastructure-async-22.9.0.jar:22.9.0]
validator-client  | ... 4 more
```

Mevboost shows the following error which is understandable (I've omitted my address): 

```
mev-boost  | time="2022-09-08T11:45:36Z" level=warning msg="error calling registerValidator on relay" error="HTTP error response: 400 / {\"code\":400,\"message\":\"not a known validator: 0x0\"}\n" method=registerValidator module=service numRegistrations=1 ua=okhttp/4.9.2 url="https://builder-relay-goerli.flashbots.net/eth/v1/builder/validators"
``` 

**Frequency:** every~5 minutes

### Versions (Add all that apply)
* Software version: Teku v22.9.0 with mevboost and nethermind
* Docker Version: 20.10.17

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,3 +21,4 @@ For information on changes in released versions of Teku, see the [releases page]
 
 ### Bug Fixes
 - Resolves an issue with public key validation.
+- Fix `/eth/v1/validator/register_validator` responding with a 400 status code and not a helpful error message in case of exceptions

--- a/data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/validator/PostRegisterValidatorTest.java
+++ b/data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v1/validator/PostRegisterValidatorTest.java
@@ -18,6 +18,7 @@ import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.verifyNoInteractions;
 import static org.mockito.Mockito.when;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 import static tech.pegasys.teku.spec.schemas.ApiSchemas.SIGNED_VALIDATOR_REGISTRATIONS_SCHEMA;
 
@@ -58,6 +59,24 @@ public class PostRegisterValidatorTest extends AbstractDataBackedRestAPIIntegrat
                 request, SIGNED_VALIDATOR_REGISTRATIONS_SCHEMA.getJsonTypeDefinition()))) {
 
       assertThat(response.code()).isEqualTo(SC_OK);
+    }
+  }
+
+  @Test
+  void shouldReturnServerErrorWhenThereIsAnExceptionWhileRegistering() throws IOException {
+    final SszList<SignedValidatorRegistration> request =
+        dataStructureUtil.randomSignedValidatorRegistrations(10);
+    when(validatorApiChannel.registerValidators(request))
+        .thenReturn(SafeFuture.failedFuture(new IllegalStateException("oopsy")));
+
+    try (Response response =
+        post(
+            PostRegisterValidator.ROUTE,
+            JsonUtil.serialize(
+                request, SIGNED_VALIDATOR_REGISTRATIONS_SCHEMA.getJsonTypeDefinition()))) {
+
+      assertThat(response.code()).isEqualTo(SC_INTERNAL_SERVER_ERROR);
+      assertThat(response.body().string()).isEqualTo("{\"code\":500,\"message\":\"oopsy\"}");
     }
   }
 

--- a/data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostRegisterValidator.java
+++ b/data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostRegisterValidator.java
@@ -13,7 +13,7 @@
 
 package tech.pegasys.teku.beaconrestapi.handlers.v1.validator;
 
-import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_GATEWAY;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_BAD_REQUEST;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_INTERNAL_ERROR;
@@ -103,7 +103,8 @@ public class PostRegisterValidator extends MigratingEndpointAdapter {
             .handle(
                 (__, error) -> {
                   if (error != null) {
-                    return AsyncApiResponse.respondWithError(SC_BAD_GATEWAY, error.getMessage());
+                    return AsyncApiResponse.respondWithError(
+                        SC_INTERNAL_SERVER_ERROR, error.getMessage());
                   }
                   return AsyncApiResponse.respondOk(null);
                 }));

